workspaces/src/worker/impls.rs

```diff
@@ -0,0 +1,163 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+use async_trait::async_trait;
+use near_crypto::InMemorySigner;
+use near_primitives::borsh::BorshSerialize;
+use near_primitives::types::{AccountId, Balance, FunctionArgs, StoreKey};
+
+use crate::network::{
+    Account, AllowDevAccountCreation, CallExecution, CallExecutionDetails, Contract, NetworkClient,
+    NetworkInfo, StatePatcher, TopLevelAccountCreator,
+};
+use crate::rpc::client::Client;
+use crate::worker::Worker;
+
+unsafe impl<T> Send for Worker<T> where T: Send {}
+unsafe impl<T> Sync for Worker<T> where T: Sync {}
+
+impl<T> Clone for Worker<T> {
+    fn clone(&self) -> Self {
+        Self {
+            workspace: self.workspace.clone(),
+        }
+    }
+}
+
+impl<T> AllowDevAccountCreation for Worker<T> where T: AllowDevAccountCreation {}
+
+#[async_trait]
+impl<T> TopLevelAccountCreator for Worker<T>
+where
+    T: TopLevelAccountCreator + Send + Sync,
+{
+    async fn create_tla(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+    ) -> anyhow::Result<CallExecution<Account>> {
+        self.workspace.create_tla(id, signer).await
+    }
+
+    async fn create_tla_and_deploy<P: AsRef<Path> + Send + Sync>(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+        wasm: P,
+    ) -> anyhow::Result<CallExecution<Contract>> {
+        self.workspace.create_tla_and_deploy(id, signer, wasm).await
+    }
+}
+
+impl<T> NetworkInfo for Worker<T>
```

Comment:
given all of the manual impls and this worker is just used to access the network, perhaps this is a use case for `Deref` and `DerefMut`. Definitely not the cleanest use case, but this feels like it could easily miss functionality or break as the lib gets built out.

wdyt @matklad ? Do you have an intuition that strongly favours yes or no here?

Reply:
My general intuition here is that all this type-and-trait-fu is not needed here. It seems simpler (for the user) if we stick with structs and inherent methods, and just return a runtime error for unsupported combinations of networks/requests:

```rust
pub struct Worker {
    rpc_addr: String,
    sandbox_process: Option<std::process::Child>,
}
```

Reply:
@matklad So, I could see us supporting custom networks in the future which is why there's all these different traits and such. `workspaces-js` already has similar stuff but under abstract classes instead, and I was trying to model off of that a little bit. 

Either ways, missing a trait impl isn't that bad here, since the compiler would just complain about it. If anything, allowing users to do `dev_deploy` on a concrete type/network that doesn't support it such as mainnet could be dangerous, because at that point, we're kinda relying on the devs to bail us out here. So the traits here would act as our separation of concern just in case something like that happens and we fall back on the compiler to help us here